yb-voyager/cmd/common.go

```diff
@@ -405,67 +402,40 @@ func CreateMigrationProjectIfNotExists(dbType string, exportDir string) {
 			utils.ErrExit("couldn't create sub-directories under %q: %v", filepath.Join(projectDirPath, "schema"), err)
 		}
 	}
-	migUUID, err := generateAndStoreMigrationUUIDIfRequired(exportDir)
-	if err != nil {
-		utils.ErrExit("couldn't generate/store migration UUID: %w", err)
-	}
 
-	err = metadb.CreateAndInitMetaDBIfRequired(exportDir)
+	initMetaDB()
+	metaDB.UpdateMigrationStatusRecord(func(record *metadb.MigrationStatusRecord) {
+		record.SourceDBConf = source
+		record.SourceDBConf.Password = ""
```

Comment:
Make this idempotent as this code path is called multiple times. Make `SourceDBConf` a pointer type. Set its value only if it is `nil`. Create a clone of `source` before assigning its address to the `SourceDBConf`.

What happens if it is a `import data file` command?

Reply:
I see. I didn't get it before because i was not using pointer with msr.Source. But now i think i will. Will check and update the code to not do anything if source is `nil`